BDArmory/Weapons/BDExplosivePart.cs

```diff
@@ -346,6 +346,12 @@ public void ParseWarheadType()
         public void DetonateIfPossible()
         {
             if (!HighLogic.LoadedSceneIsFlight || part == null) return;
+            var MF = VesselModuleRegistry.GetModule<MissileFire>(vessel, true);
```

Comment:
This probably needs to be set when the part containing this is fired or armed instead of here, otherwise the team string won't get assigned until this tries to detonate and the fratricide checks from other detonating parts won't see this as being the same team.